src/ui/state/manager.rs

```diff
@@ -65,7 +67,15 @@ impl ManagerHomeState {
     }
 
     pub fn update_spend_txs(&mut self, txs: Vec<model::SpendTx>) {
-        self.spend_txs = txs.into_iter().map(SpendTransactionListItem::new).collect();
+        self.spend_txs = txs
+            .into_iter()
+            .filter(|tx| {
+                tx.deposit_outpoints
+                    .iter()
+                    .all(|outpoint| self.spendable_outpoints.contains(&outpoint))
+            })
+            .map(SpendTransactionListItem::new)
+            .collect();
```

Comment:
we have the problem where self.spendable_outpoints is empty because list_vaults result is not yet arrived
I think the solution would be to filter again in `update_vaults` the spend_txs

Reply:
You are right, thanks. I updated the MR but it's quite full of `clone` now :sob: